presto-main-base/src/main/java/com/facebook/presto/sql/relational/FunctionResolution.java

```diff
@@ -302,7 +302,10 @@ public boolean isTryFunction(FunctionHandle functionHandle)
 
     public boolean isFailFunction(FunctionHandle functionHandle)
     {
-        return functionAndTypeResolver.getFunctionMetadata(functionHandle).getName().equals(functionAndTypeResolver.qualifyObjectName(QualifiedName.of("fail")));
+        // todo: Revert this hack once constant folding support lands in C++.
```

Comment:
this seems a bit sketchy -- why only the fail function and not the other functions in this class? Seems like it's that you happen to know this function is only used in the rowExpressionInterpreter. 

 I would rather add a new method to this class isJavaBuiltinFailFunction or something so that the assumption that something uses the  java built in namespace is isolated to the RowExpressionInterpreter.